packages/foundation-models/src/azure-openai/azure-openai-chat-completion-stream-response.ts

```diff
@@ -0,0 +1,44 @@
+import type { AzureOpenAiCompletionUsage } from './client/inference/schema/index.js';
+import type { AzureOpenAiChatCompletionStream } from './azure-openai-chat-completion-stream.js';
+
+/**
+ * Azure OpenAI chat completion stream response.
+ */
+export class AzureOpenAiChatCompletionStreamResponse<T> {
+  private _usage: AzureOpenAiCompletionUsage | undefined;
+  private _finishReason: string | undefined;
+  private _stream: AzureOpenAiChatCompletionStream<T> | undefined;
+
+  public get usage(): AzureOpenAiCompletionUsage {
+    if (!this._usage) {
+      throw new Error('Response stream is undefined.');
+    }
+    return this._usage;
+  }
+
+  public set usage(usage: AzureOpenAiCompletionUsage) {
+    this._usage = usage;
+  }
+
+  public get finishReason(): string {
```

Comment:
We could consider:

```suggestion
  public get finishReason(index = 0): string {
```

Technically, the finish reason might differ for different indices. But I doubt this is a frequent use case and we could always add this later, as this would be a none-breaking change, right? If so, I'd leave it out for now..

Reply:
This was probably a wrong assumption then. Atleast for me, the docs were not explicitly clear on this. @MatKuhr can you please confirm? We can fix that in a separate PR.

Reply:
In Java we don't consider n>1 for the convenience API. If users want to access anything with index != 0, they have to do it "manually". See [this code](https://github.com/SAP/ai-sdk-java/blob/main/foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/model/OpenAiChatCompletionDelta.java#L32).

Reply:
Issue created -> https://github.tools.sap/AI/gen-ai-hub-sdk-js-backlog/issues/173